src/LegacySubscriptions/ServiceProvider.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Give\LegacySubscriptions;
+
+use Closure;
+use Give\ServiceProviders\ServiceProvider as ServiceProviderInterface;
+
+/**
+ * Class ServiceProvider - LegacySubscriptions
+ *
+ * This handles the loading of all the legacy codebase included in the LegacySubscriptions /includes directory.
+ * DO NOT EXTEND THIS WITH NEW CODE as it is intended to shrink over time as we migrate over
+ * to the new ways of doing things.
+ *
+ * @unreleased
+ */
+class ServiceProvider implements ServiceProviderInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function register()
+    {
+        $recurringIsInstalled = defined('GIVE_RECURRING_VERSION') && GIVE_RECURRING_VERSION;
+        $recurringMeetsRequirements = $recurringIsInstalled && GIVE_RECURRING_VERSION > '1.14.1';
+
+        if ($recurringMeetsRequirements || !$recurringIsInstalled) {
+            $this->includeLegacyFiles();
+            $this->bindClasses();
+        }
+
+        $this->includeLegacyHelpers();
```

Comment:
@jonwaldstein Why isn't this included in the above condition?

Reply:
So this file basically takes `give_recurring()->is_recurring`, `give_recurring()->is_donation_recurring`, and `give_recurring()->get_period` and pulls them in as functions that don't rely on `give_recurring()`.  This way for the time being, we don't need to use the main give_recurring class to figure out if a donation is recurring.